rules/DeadCode/Rector/If_/RemoveUnusedNonEmptyArrayBeforeForeachRector.php

```diff
@@ -156,4 +161,35 @@ private function isUselessBooleanAnd(BooleanAnd $booleanAnd, Expr $foreachExpr):
 
         return $this->countManipulator->isCounterHigherThanOne($booleanAnd->right, $foreachExpr);
     }
+
+    private function refactorStmtsAware(StmtsAwareInterface $stmtsAware): ?StmtsAwareInterface
+    {
+        foreach ((array) $stmtsAware->stmts as $key => $stmt) {
```

Comment:
It seems it can be optimized by only get `$stmtsAware->stms[$lastKey - 2]` and `$stmtsAware->stms[$lastKey - 1]`, that make loop is not needed

Reply:
New phpunit https://github.com/sebastianbergmann/phpunit/releases/tag/10.1.0  seems cause error on paratest, the handling is still on progress https://github.com/paratestphp/paratest/pull/752

I will set phpunit to 10.0.19 temporary.

Reply:
I temporary set phpunit to 10.0.19 https://github.com/rectorphp/rector-src/pull/3611/commits/e23e2e372bb0eeed9bbcaa16f3b6606b8cc1d91d while wait for paratest handling for it.

Reply:
paratest 7.1.3 just released https://github.com/paratestphp/paratest/releases/tag/v7.1.3, let's try https://github.com/rectorphp/rector-src/pull/3611/commits/6091029c75678ca145c329ab9ebe07d8304ab4bd